src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -79,27 +79,31 @@ export class CodeButtonComponent implements OnInit, OnChanges {
         private ideSettingsService: IdeSettingsService,
     ) {}
 
-    ngOnInit() {
-        this.accountService
-            .identity()
-            .then((user) => {
-                this.user = user!;
-                this.refreshTokenState();
-
-                this.copyEnabled = true;
-                this.useSsh = this.localStorage.retrieve('useSsh') || false;
-                this.useToken = this.localStorage.retrieve('useToken') || false;
-                this.localStorage.observe('useSsh').subscribe((useSsh) => (this.useSsh = useSsh || false));
-                this.localStorage.observe('useToken').subscribe((useToken) => (this.useToken = useToken || false));
-
-                if (this.useSsh) {
-                    this.useSshUrl();
-                }
-                if (this.useToken) {
-                    this.useHttpsUrlWithToken();
-                }
-            })
-            .then(() => this.loadParticipationVcsAccessTokens());
+    async ngOnInit() {
+        const user = await this.accountService.identity();
+        if (!user) {
+            return;
+        }
+        this.user = user;
+
+        this.refreshTokenState();
+
+        this.copyEnabled = true;
+        this.useSsh = this.localStorage.retrieve('useSsh') || false;
+        this.useToken = this.localStorage.retrieve('useToken') || false;
+        this.localStorage.observe('useSsh').subscribe((useSsh) => {
```

Comment:
Why do we need the `retrieve` first and then the `observe` again? If it is really necessary, can you add a comment so that future developers also know why this is required?

Reply:
Turns out we actually do not need the observer+subscribe at all, as switching between the links is works synchronously. I found another race condition though, so I moved the  two method calls into the subscribe block of the profileService.